### PR TITLE
New Bitbucket status Spice

### PR DESCRIPTION
**Pull request by qqq3**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.

- [x] New Instant Answer: **`New {IA Name} Instant Answer`**
- [ ] Improvement
    - [ ] Bug fix: **`{IA Name}: Fix {Issue number or one-line description}`**
    - [ ] Enhancement: **`{IA Name}: {Description of Improvements}`**
- [ ] Non–Instant Answer
    - [ ] Other (Role, Template, Test, Documentation, etc.): **`{SpiceRole/Templates/Tests/Docs}: {Short Description}`**

###### Description of changes
Return Bitbucket status

---

Instant Answer Page: https://duck.co/ia/view/bitbucket_status

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @qqq3

### CODE REVIEW

**Comment by daxtheduck** [comment]
@qqq3 Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please head to https://quackslack.herokuapp.com/ to get an invite.

>More Info: http://docs.duckduckhack.com/submitting/submitting-overview.html

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Bitbucket status)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/bitbucket_status_mobile.png?HRhHQ8wz6g)

**Comment by zekiel** [comment]
@qqq3 Looks and works great! Do you think it should include triggering on things like "status of bitbucket" or https://duckduckgo.com/?q=is+bitbucket+down&t=ffab&ia=answer since that IA is currently covering only the website?

**Comment by qqq3** [comment]
@zekiel Ok, will include more triggers

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Bitbucket status)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/bitbucket_status_mobile.png?zj49Ug5cxO)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Bitbucket status)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/bitbucket_status_mobile.png?BAY63mNZHY)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Bitbucket status)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/bitbucket_status_mobile.png?c1lqjRe5Q4)

**Comment by zekiel** [comment]
thx for the updates @qqq3! I'll need some help from @moollaza here bc the isitup Instant Answer seems to overtake this on queries like https://beta.duckduckgo.com/?q=is+bitbucket+down%3F&ia=answer

**Comment by moollaza** [comment]
> I'll need some help from @moollaza here bc the isitup Instant Answer seems to overtake this on queries like https://beta.duckduckgo.com/?q=is+bitbucket+down%3F&ia=answer

We can add some negative triggers to the IsItUp Spice if we want to show the various "Status" IA's for those kinds of queries. It looks like the GitHub Status IA doesn't trigger on that query currently, so we're showing IsItUp: https://duckduckgo.com/?q=is+github+up%3F&ia=answer

If we feel this should be changed, we can update separately. This IA looks ready for merge to me 👍

**Comment by daxtheduck** [comment]
@qqq3 this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

File: share/spice/bitbucket_status/bitbucket_status.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        }
+        
+        var dataDetails = {
+            status: api_result.status.indicator,
+            description: api_result.status.description
+        };
+
+        Spice.add({
+            id: "bitbucket_status",
+            name: "Status",
```
Comment (moollaza) [inline share/spice/bitbucket_status/bitbucket_status.js]: @qqq3 whoops, let's change this to `Answer`

Comment (moollaza) [inline share/spice/bitbucket_status/bitbucket_status.js]: I'll update the IA Page accordingly.

File: share/spice/bitbucket_status/bitbucket_status.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            data: dataDetails,
+            signal: 'low',
```
Comment (moollaza) [inline share/spice/bitbucket_status/bitbucket_status.js]: This shouldn't be necessary. It will force other IAs to take over the query

Comment (qqq3) [inline share/spice/bitbucket_status/bitbucket_status.js]: @moollaza Need remove `signal: 'low'` or not?

Comment (moollaza) [inline share/spice/bitbucket_status/bitbucket_status.js]: Yes, please do

**Pull request merged**